### DISCUSSION TITLE
Suspend/Unsuspend Users

### DISCUSSION THREAD

**Discussion by mp21-org** [btcpayserver/btcpayserver#2812] [Ideas / Feature Requests]
I'm currently in the process of creating a provisioning module for btcpay in WHMCS.
The idea is rather simple: When someone orders a BTCPay Account, a new user gets created via the API and the details like username and password get emailed. (regular user registration on btcpay is disabled, only admin can create new users)

However for the automation to make sense, it would be useful to have a way of suspending users that are overdue on their billing and to unsuspend them once they are back on track.

What would a suspension of a user mean:

- all of his Greenfield API keys and bitpay/legacy API tokens become deactivated
- login to btcpay gets blocked / perhaps with a customizable message where I can direct them to the WHMCS clientarea so that they can pay their overdue bills

I think even if someone is not integrating with WHMCS but offers a paid BTCPay Service, this feature would be useful.

Happy to hear your oppinions.

**Comment by woutersamaey** [comment]
This seams like a valid use case if you ask me.

**Reply by dstrukt** [reply]
+1

**Comment by woutersamaey** [comment]
I would propose we add a SuspendedFrom timestamp on the user's account. If it is NULL or in the future, the user is currently not suspended.
By choosing a timestamp, we can also program users to expire/suspend in the future. I.e. they have a subscription and their account expires in 30 days.

Should be exposed in the Greenfield API as well.
